### DISCUSSION TITLE
No endpoint for validating webhook

### DISCUSSION THREAD

**Discussion by anekdoti** [dotnet/dotnet-operator-sdk#290] [Q&A] [answered]
I have implemented a validating webhook in my operator. It seems to be correctly registered according to the logs, and the Kubernetes resources for Service and ValidatingWebhookConfiguration look fine. Still, when applying a CR to the cluster, I get

    Error from server (InternalError): error when creating ".\\example-cr.yaml": Internal error occurred: failed calling webhook "{{my-validator}]": Post https://{{my-service-url}}/{{my-path-to-validator}}/validate?timeout=10s: no endpoints available for service "{{my-service}}"

Is there anything I should look into at first? Thank you very much!

**Comment by erin-allison** [comment]
I'm pretty sure this should be an issue, not a discussion, but here are some questions to start:

Is your pod coming to a ready state? Does the pod definition have blocks for readiness and liveness checks but maybe they weren't configured in your operator?

Do the selectors on your service match your pod(s)?

Can you share your config files for further examination?

**Comment by buehler** [comment]
Hey @anekdoti 

Is the service being created in Kubernetes?
There should exist a service that points to your pod.

**Reply by jestinjo** [reply]
I have implemented the webhook validation for the operator, however there is no endpoint attached to it neither is there a service running. It gives a similar error to the error mentioned above. If I remove the manual webhook registration from the Program.cs, operator works fine and i can apply the CRs. Im pretty new to the whole idea so please bear with me if i ask stupid questions. 

Thanks

**Comment by anekdoti** [comment, marked as answer]
Sorry for not replying earlier! It was actually a combination of all the points mentioned by you and works fine now :-) Thank you a lot for the suggestions.

**Reply by buehler** [reply]
No worries ;-)
